src/recipient-recieved-scanner.ts

```diff
@@ -127,12 +127,40 @@ export class RecipientRecievedScanner {
         }
       : distributorsData.distributors;
 
+    // Only apply error isolation when processing multiple distributors
+    const distributorCount = Object.keys(distributorsToProcess).length;
+
+    if (distributorAddress || distributorCount === 1) {
+      // Single distributor - no error isolation
+      await this.processSingleDistributor(
+        distributorsToProcess,
+        blockNumbersData,
+        yesterdayStr,
+      );
+    } else {
+      // Multiple distributors - apply error isolation
+      await this.processMultipleDistributors(
+        distributorsToProcess,
+        blockNumbersData,
+        yesterdayStr,
+      );
+    }
+  }
+
+  /**
+   * Processes a single distributor without error isolation.
+   * @private
+   */
+  private async processSingleDistributor(
```

Comment:
Lets combine these two methods internall and just pass an optional arg of the distributor address to filter.